src/wcs/src/main/java/org/vfny/geoserver/util/WCSUtils.java

```diff
@@ -878,4 +878,43 @@ private static GridGeometry2D simpleGridGeometryFit(
             throw new RuntimeException("Failed to invert grid to world", e);
         }
     }
+
+    /**
+     * Alternative to {@link CoverageInfo#boundingBox()} used to determine native bounds prior to
+     * projection policy being applied.
+     *
+     * <p>This method allows {@link CoverageInfo#getNativeCRS()} to override native bounding box
+     * srs.
+     *
+     * <p>This differs from {@link CoverageInfo#boundingBox()} in that the native bounds returned
+     * are only determined using native bounding box and native crs information. It is not willing
+     * to back-transform from lat lon bounding box. It also unwilling to grow the native bounding
+     * box while re-projecting to declared crs.
+     *
+     * @param meta CoverageInfo configuration
+     * @return Native bounds, using native bounding box crs
+     * @throws IllegalStateException if coverage info does not define a native bounding box with
+     *     native crs information
+     */
+    public static ReferencedEnvelope toNativeBounds(CoverageInfo meta) {
```

Comment:
If at all possible I'd avoid code duplication and improve ResourceInfo.boundingBox instead (which also has much larger exposure to the code base, thus better overall testing)